interfaces/builtin/storage_framework.go

```diff
@@ -0,0 +1,153 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const storageFrameworkPermanentSlotAppArmor = `
+# Description: Allow use of aa_is_enabled()
+
+/sys/module/apparmor/parameters/enabled r,
+@{PROC}/@{pid}/mounts                   r,
+/sys/kernel/security/apparmor/.access   rw,
+
+# Description: Allow owning the registry and storage framework bus names on the session bus.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (send)
+    bus=session
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={RequestName,ReleaseName,GetConnectionCredentials}
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (bind)
+    bus=session
+    name=com.canonical.StorageFramework.Registry,
+
+dbus (bind)
+    bus=session
+    name=com.canonical.StorageFramework.Provider.*,
+`
+
+const storageFrameworkConnectedSlotAppArmor = `
+# Description: Allow clients to access the registry and storage framework services.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Registry
+    path=/com/canonical/StorageFramework/Registry
+    peer=(label=###PLUG_SECURITY_TAGS###),
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Provider.*
+    path=/provider/*
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const storageFrameworkConnectedPlugAppArmor = `
+# Description: Allow access to the registry and storage framework services.
+
+#include <abstractions/dbus-session-strict>
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Registry
+    path=/com/canonical/StorageFramework/Registry
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive, send)
+    bus=session
+    interface=com.canonical.StorageFramework.Provider.*
+    path=/provider/*
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`
+
+type StorageFrameworkInterface struct{}
+
+func (iface *StorageFrameworkInterface) Name() string {
+	return "storage-framework"
```

Comment:
Storage framework is a generic name. Is there any documentation for what the canonical storage framework is doing?